### PR TITLE
[REVIEW] Fix issue with column and scalar re-assignment 

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #10305  This PR fixes issues where the `scalar` that is being assigned was being type-casted to the column's type. Added test coverage for the same.

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -1107,6 +1107,7 @@ def __getitem__(self, arg):
     @annotate("DATAFRAME_SETITEM", color="blue", domain="cudf_python")
     def __setitem__(self, arg, value):
         """Add/set column by *arg or DataFrame*"""
+        # import pdb;pdb.set_trace()
```
Comment (bdice) [inline python/cudf/cudf/core/dataframe.py:1110]:
```suggestion
```

**Review by bdice — APPROVED** [review]

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py]: oops, sorry to miss this.

Comment (bdice) [reply python/cudf/cudf/core/dataframe.py]: It slipped by without a trace. 😉

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**